tests/test_valcode_purity.py

```diff
@@ -0,0 +1,40 @@
+from ethereum import utils
+from ethereum.tools.tester import TransactionFailed
+import pytest
+
+def test_deposit_succeeds_with_pure_sig_validator(casper, funded_privkey,
```

Comment:
Would probably be better to implement this more generically with `pytest.mark.parametrize`. See `test_voting` for an example https://github.com/ethereum/casper/blob/master/tests/test_voting.py#L7